### ISSUE TITLE
Upgrade GLM (and possibly add GLI?)

### ISSUE THREAD

**Issue by totalgee** [cinder/Cinder#1794]
Hi there, I was wondering if there was any plan to use a newer version of GLM in the upcoming Cinder (0.9.1?) release.  I was trying to use GLI (http://gli.g-truc.net/0.8.2/index.html) in my project, but this requires a newer version of GLM (0.9.8.0, which it includes in its "external" directory), e.g. for all the SIMD stuff.  *(Incidentally, it might be nice for Cinder to use GLI for its KTX and DDS loading (and saving) stuff.)*

Not surprisingly, mixing my project built with an updated GLM and Cinder 0.9.0 (without recompiling the libs) causes a crash on startup...I get a `cinder::app::ExcInvalidWindow` exception thrown, probably because memory is out of whack.  I know it will probably work if I rebuild all of Cinder with a newer GLM, but I was trying to avoid that for this project if possible (it's easier if other contributors on the project can just install the "official" version from the website).

Thanks,
Glen.

**Comment by andrewfb** [comment]
This is on the list for immediately _after_ 0.9.1 but for stability reasons isn't something we're likely to change before. Incidentally, Cinder does have the ability to load both KTX and DDS files, though it's not quite as robust as GLI. It's something I've considered moving us to in the future - is there specific functionality you're requiring that Cinder lacks currently?

**Comment by totalgee** [comment]
In particular, I was looking for the functionality to **save** a KTX file (a mipmapped 2D array texture, in particular), so I could load it in another (non-Cinder) program that uses GLI.

Also, there are some limitations, such as array textures [are not supported](https://github.com/cinder/Cinder/blob/dev_0.9.2/src/cinder/gl/TextureFormatParsers.cpp#L66-L67) by the Cinder KTX loader at the moment:

```c++
	if( header.numberOfArrayElements != 0 )
		throw KtxParseExc( "Array textures not currently supported" );
```

**Comment by andrewfb** [comment]
OK - that makes sense. I will put more GLI investigation on my 0.9.2 list. Sorry we don't have a better solution in the immediate - perhaps a slightly older version of GLI? It looks like its last update referencing a new version of GLM was 0.8.1.1, so perhaps even 0.8.1 would work?

**Comment by totalgee** [comment]
Even going back to GLI 0.6.0, the version of GLM was 0.9.7, and Cinder 0.9.0 uses GLM 0.9.6.3.  Maybe it will work (have fewer dependencies), though; I'll give it a try.

**Comment by totalgee** [comment]
Nope, no luck... GLI 0.7.0 was the one that added KTX load/save support.  And with that version (and the corresponding GLM 0.9.7.1) it still crashes.  And if I try to use the existing Cinder GLM (0.9.6.3) with that version of GLI, it won't compile; it requires at least 0.9.7.  Oh well, guess I'll take a different approach for now!

**Comment by richardeakin** [comment]
We can move forward with upgrading GLM to latest in the _dev_0.9.2_ branch, if that helps keep things moving. This branch will be merged into master as soon as we can post 0.9.1 release.

**Comment by totalgee** [comment]
Cool, thanks.  In the meantime, I made a workaround.
